File: frogment/src/main/java/ch/grze/frogment/frogment/FrogmentComponent.java
package ch.grze.frogment.frogment;

import ch.grze.frogment.activity.FrogmentActivityInterface;
import ch.grze.frogment.core.Core;

public final class FrogmentComponent {
    private final FrogmentInterface frogment;
    private final Core core;

    private FrogmentData data;

    public FrogmentComponent(Core core, FrogmentInterface frogment) {
        this.core = core;
        this.frogment = frogment;
    }

    public FrogmentData getData() {
        return data;
    }

    public void setData(FrogmentData data) {
        this.data = data;
    }

    public FrogmentActivityInterface getFrogmentActivity() {
        return (FrogmentActivityInterface) frogment.getActivity();
    }

    public <T> T getTypedActivity() {
        return (T) frogment.getActivity();
    }
}

File: frogment/src/main/java/ch/grze/frogment/frogment/FrogmentComponent.kt
package ch.grze.frogment.frogment

import ch.grze.frogment.activity.FrogmentActivityInterface
import ch.grze.frogment.core.Core
import ch.grze.frogment.core.navigation.FrogmentData
import ch.grze.frogment.core.navigation.Navigator
import ch.grze.frogment.core.navigation.NavigatorAware

class FrogmentComponent : NavigatorAware, FrogmentActivityAware, FrogmentDataAware {
    override lateinit var data: FrogmentData
    lateinit var core: Core
    lateinit var frogment: FrogmentInterface

    override val navigator: Navigator by lazy {
        getFrogmentActivity().navigator
    }

    override fun getFrogmentActivity(): FrogmentActivityInterface = frogment.activity as FrogmentActivityInterface

    override fun <T> getTypedActivity(): T = frogment.activity as T
}